src/Mvc/Mvc.ViewFeatures/src/Buffers/ViewBufferTextWriter.cs

```diff
@@ -109,6 +109,11 @@ public override void Write(char[] buffer, int index, int count)
             {
                 throw new ArgumentNullException(nameof(buffer));
             }
+            
+            if (index == 0 && count == 0)
```

Comment:
Not sure of the exact approach / level of complexity we'd like to add here. We could do something a bit fancier like:

```suggestion
            if (index >= 0 && index < buffer.Length && count == 0)
```

Reply:
We could simply copy the checks from here: https://github.com/dotnet/runtime/blob/7938f9d9fadc3952d792329be602668bef124f23/src/libraries/System.Private.CoreLib/src/System/IO/TextWriter.cs#L159

Reply:
Thanks, updated & added a test.